public/main/inc/lib/webservices/Rest.php

```diff
@@ -1432,6 +1435,97 @@ public function subscribeUserToCourse($params)
         return [true];
     }
 
+     /**
+     * @param $course_id
+     * @return array
+     */
+    public function getHomework($course_id){
+        //echo "string";
+        $workTable =Database::get_course_table(TABLE_STUDENT_PUBLICATION);
+        $workassignment =Database::get_course_table(TABLE_STUDENT_PUBLICATION_ASSIGNMENT);
+        $homeWork=[];
+        $idUser=$this->user->getId();
+            
+        $count_homework="SELECT *,w.id as id_w,(SELECT AVG(if(qualification>0,qualification,NULL))  
+                    FROM $workTable WHERE c_id=$course_id AND active in (0,1) AND filetype='file'
+                    AND user_id= $idUser AND parent_id = w.id) AS cal_tarea 
+                    FROM $workTable as w 
+                    INNER JOIN $workassignment AS a ON a.publication_id=w.id
+                    WHERE w.parent_id=0 AND w.active IN (1,0) AND w.c_id = $course_id;";
+
+        $result = Database::query($count_homework);  
+        //url work
+        $course_code=self::getCourseCode($course_id); 
+        if (Database::num_rows($result)) {
+                while($row= Database::fetch_array($result, 'ASSOC')){
+                    $id=$row["id_w"];
+                    $url="https://".$_SERVER['SERVER_NAME']."/main/work/work_list.php?cidReq=$course_code&id_session=0&gidReq=0&gradebook=0&origin=&id=$id";
```

Comment:
Please use api_get_path(WEB_CODE_PATH) to obtain the equivalent of "http://$_SERVER['SERVER_NAME']/main/"